docs/terrain.md

```diff
@@ -0,0 +1,123 @@
+# Roblox Terrain Binary Format
+
+This document describes the Terrain binary format. In this format there is no field for Version, so it is assumed that any future changes will be additions to the existing format or a new format entirely. This specification does not include the adjacent PhysicsGrid binary format, only SmoothGrid.
+
+# Contents
+
+- [Document Conventions](#document-conventions)
+- [File Structure](#file-structure)
+- [Data Types](#data-types)
+  - [Chunk](#chunk)
+  - [Voxel](#voxel)
+  - [Voxel.Flag](#voxel.flag)
+  - [Material](#material)
+
+## Document Conventions
+
+This document assumes a basic understanding of Rust's convention for numeric types. For example:
+
+- `u32` is an unsigned 32-bit integer
+- `f32` is a 32-bit floating point number
+
+All numeric types are little endian. Floats are stored as a dividend of their type's numeric maximum; as such, a `f8` with a value of `0x31` translates to `0x31 / 0xFF`, or `~0.192`. Floats are unsigned. Signed integers are stored with their absolute value, with preceding bytes containing the signedness information; meaning the range for an `i8` is `-0xFF` -> `0xFF`, instead of `-0x7F` to `0x7F`.
```

Comment:
The conventions here are a bit confusing. To me, the name `i8` suggests the number is 1 byte, but if the sign is stored in the previous byte, that seems to be 2 bytes. Can you give an example of what that looks like in practice?